### PR TITLE
Documentation for setting `dtype=str` before passing dataframe to `create_from_dataframe`

### PR DESCRIPTION
**Pull request by abafzal**
# ↪️ Pull Request

## ✔️ PR Todo

- [N/A] Added/updated testing for this change
- [x] Included links to related issues/PRs, Relates to #323 
- [x] Update relevant [docs](https://github.com/Datatamer/tamr-client/tree/master/docs) + docstrings
- [x] Update the [CHANGELOG](https://github.com/Datatamer/tamr-client/blob/master/CHANGELOG.md) under the current `-dev` version

### CODE REVIEW

**Comment by abafzal** [comment]
@pcattori i dont know why i cant add you as a reviewer to the PR directly but here you go i am tagging you

**Comment by pcattori** [comment]
@abafzal could you change `[x]` to `N/A` for items that don't apply (e.g. testing)?

Additionally, can you add something like "Relates to #323" (that's what the second checkbox "Included links to related issues/PRs" is for 😛 )

File: CHANGELOG.md [hunk 1]
```diff
@@ -33,6 +33,7 @@
 
   **BUG FIXES**
   - Links from our docs to the `requests` docs were outdated. Links have been updated to point to the new `requests` docs URL.
+  - [#323](https://github.com/Datatamer/tamr-client/issues/323) <tamr.datasets.create_from_dataframe should use dtype=str to avoid dataset upload error>
```
Comment (pcattori) [inline CHANGELOG.md]: Replace description with a more user-facing one:

```suggestion
  - [#323](https://github.com/Datatamer/tamr-client/issues/323) Documentation for setting `dtype=str` before calling `client.datasets.create_from_dataframe`
```

File: docs/user-guide/spec.md [hunk 2]
```diff
@@ -59,8 +59,8 @@ This makes it easy to create a Tamr dataset from a CSV:
 ```python
 import pandas as pd
 
-df = pd.read_csv("my_data.csv")
-dataset = tamr.datasets.create_from_dataframe(df, "primary key name", "My Data")
+df = pd.read_csv("my_data.csv", dtype=str)
```
Comment (pcattori) [inline docs/user-guide/spec.md]: Can you include some prose (either as a comment in the codeblock or outside the codeblock) explaining the need for this? Even one sentence is good!

Comment (abafzal) [reply docs/user-guide/spec.md]: done

**Pull request merged**